.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+name: Build
+
+on:
+  [push]
+
+jobs:
+  build:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    outputs:
+      # Get from step "setversion"
+      artifact-version: ${{ steps.setversion.outputs.version }}
+    env:
+      BUILD_VERSION: SNAPSHOT
+      # Name of the project; used in multiple steps (e.g. for upload)
+      PROJECT_NAME: webdav-nio-adapter
+    steps:
+      # Foreign Action: Checkout the current commit
+      - uses: actions/checkout@v2
+      # Foreign Action: Setup Java Runtime Environment
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 11
+          server-id: bintray-jcenter
+          server-username: BINTRAY_USERNAME # Defined in step "deploytojcenter"
+          server-password: BINTRAY_API_KEY # See above
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Ensure to use tagged version
+        id: taggedVersion
+        run: mvn versions:set --file ./pom.xml -DnewVersion=${GITHUB_REF##*/}
+        if: startsWith(github.ref, 'refs/tags/') # Run if ref is tagged (e.g. "v.1.4")
+      - name: Export the project version to the job environment and fix it as an ouput of this step
+        id: setVersion # Output used above (outputs.artifact-version)
+        run: |
+          v=$(mvn help:evaluate "-Dexpression=project.version" -q -DforceStdout)
+          echo "::set-env name=BUILD_VERSION::${v}"
+          echo "::set-output name=version::${v}"
+      - name: Build and Test
+        id: buildAndTest
+        run: |
+          mvn clean test jacoco:report verify -Pcoverage
+          cp target/site/jacoco/jacoco.xml jacoco.xml
+      - name: Run Codacy-coverage-reporter
+        id: coverageReporter
+        uses: codacy/codacy-coverage-reporter-action@master
+        with:
+          project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+          coverage-reports: jacoco.xml
+        env:
+          CODACY_REPORTER_VERSION: 7.1.0
+      - name: Install Artifact
+        id: installArtifact
+        run: mvn -B install
+      - name: Upload snapshot artifact ${{ env.PROJECT_NAME }}-${{ env.BUILD_VERSION }}.jar
+        id: uploadSnapshot
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.PROJECT_NAME }}-${{ env.BUILD_VERSION }}.jar
+          path: target/${{ env.PROJECT_NAME }}-*.jar
+      - name: Deploy to jcenter
+        id: deployToJcenter
+        run: mvn -B deploy
```

Comment:
Same rationale: Maven will re-run all steps. Still this needs to be a separate CI step for obvious reasons, but the least we can do is to run `mvn -B deploy -DskipTests`.